test/sim/abilities/pressure.js

```diff
@@ -85,6 +85,50 @@ describe('Pressure', function () {
 		assert.equal(battle.p2.active[0].getMoveData(Dex.getMove('snatch')).pp, 12);
 		assert.equal(battle.p2.active[1].getMoveData(Dex.getMove('imprison')).pp, 12);
 	});
+
+	it.skip(`should deduct additional PP from Max Moves`, function () {
+		battle = common.createBattle([[
+			{species: 'wynaut', moves: ['darkpulse']},
+		], [
+			{species: 'absol', ability: 'pressure', moves: ['sleeptalk']},
+		]]);
+		battle.makeChoices('move darkpulse dynamax', 'auto');
+		assert.equal(battle.p1.active[0].getMoveData(Dex.getMove('darkpulse')).pp, 22);
+	});
+
+	it.skip(`should deduct additional PP from Z-Moves`, function () {
+		battle = common.gen(7).createBattle([[
+			{species: 'wynaut', item: 'darkiniumz', moves: ['darkpulse']},
+		], [
+			{species: 'absol', ability: 'pressure', moves: ['sleeptalk']},
+		]]);
+		battle.makeChoices('move darkpulse zmove', 'auto');
+		assert.equal(battle.p1.active[0].getMoveData(Dex.getMove('darkpulse')).pp, 22);
+	});
+
+	it.skip(`should deduct additional PP from submoves that target Pressure`, function () {
```

Comment:
> So, you're saying that if you use a self-targeting move that calls moves, and that move calls a move that then targets an opposing Pressure user, then additional PP is deducted from the base move?

Yes. The move chosen by Sleep Talk, Metronome, etc. are resolved prior to target being determined. In fact, Pressure is how we know that's the case.

> What about Nature Power? Currently I think the sim might already deduct additional PP.

Yeah, Pressure should deduct an additional PP if targeted by a move called from Nature Power.

> Also, what happens to Pledge moves?

Both the interrupted Pledge move and Pledge move proper consume 2 PP when used into Pressure.

Reply:
The Nature Power one is confusing as in Gen 6 it targets normally despite being a move that calls a submove (those normally target yourself). I don't know whether it's possible for it to call a spread move (I think it would require a double or multi battle in a volcano) as that would get very confusing for PP deduction purposes.

Reply:
I don't think it'd be any more unusual than it is now. Just like Sleep Talk -> Thunderbolt can update the target from self to opponent, so Nature Power -> Lava Plame would update the targets to each Pokemon on the field. Hopefully we don't have to worry about Volcano battles anytime soon! :P